src/RepayRewardEscrow.sol

```diff
@@ -0,0 +1,215 @@
+    event BuyDeadlineUpdated(uint256 newDeadline);
+    event NewPendingGov(address indexed newOperator);
+    event OperatorChanged(address indexed oldOperator, address indexed newOperator);
+    event GovChanged(address indexed oldGov, address indexed newGov);
+    event DolaRepayment(uint256 amount);
+    event Sweep(address indexed token, uint256 amount);
+    event VestingStarted(uint256 redemptionTimestamp);
+
+    constructor(address _gov, address _operator) ERC20("lsINV", "lsINV") {
+        gov = _gov;
+        operator = _operator;
+        sweepTimestamp = block.timestamp + 365 days; // 1 year from deployment
+    }
+
+    modifier onlyOperator() {
+        require(msg.sender == operator, "Only operator");
+        _;
+    }
+
+    modifier onlyGov() {
+        require(msg.sender == gov, "Only governance");
+        _;
+    }
+
+    /**
+     * @notice Allows users to buy lsINV tokens using DOLA. lsINV can be redeemed for sINV after the redemption period.
+     * @dev The user must have a allocation set for the exact amount of DOLA they can spend and approve this contract to spend it.
+     * @return lsInvOut The number of lsINV shares purchased.
+     */
+    function buy() external returns (uint256 lsInvOut) {
+        require(block.timestamp <= buyDeadline, "Buy period ended or not started");
+        require(dolaAllocations[msg.sender] != 0, "No DOLA allocation set for user");
+
+        uint256 dolaAmountIn = dolaAllocations[msg.sender];
+
+        DOLA.safeTransferFrom(msg.sender, address(this), dolaAmountIn);
+        uint256 invAmount = dolaAmountIn * 1 ether / INV_PRICE;
+        INV.safeTransferFrom(gov, address(this), invAmount);
+        INV.approve(address(sINV), invAmount);
+
+        uint256 expectedShares = sINV.previewDeposit(invAmount);
+        lsInvOut = sINV.deposit(invAmount, address(this));
+        require(lsInvOut >= expectedShares, "Insufficient shares received");
+
+        _mint(msg.sender, lsInvOut);
+
+        dolaAllocations[msg.sender] = 0;
+
+        emit Buy(msg.sender, dolaAmountIn, invAmount, lsInvOut);
+    }
+
+    /**
+     * @notice Allows users to redeem their lsINV shares for sINV tokens after the redemption period.
+     * @dev Users must wait until the redemption timestamp to redeem their shares.
+     * @param lsInvAmount The number of lsINV shares to redeem.
+     */
+    function redeem(uint256 lsInvAmount) external {
+        require(block.timestamp >= redemptionTimestamp && redemptionTimestamp != 0, "Redemption not started yet");
+        require(lsInvAmount > 0, "lsInvAmount must be greater than zero");
+
+        _burn(msg.sender, lsInvAmount);
+
+        sINV.safeTransfer(msg.sender, lsInvAmount);
+
+        emit Redeem(msg.sender, lsInvAmount);
+    }
+
+    /**
+     * @notice Allows anyone to send DOLA to the sale handler for debt repayment.
+     * Will send up to the Sale Handler capacity.
+     */
+    function sendToSaleHandler() external {
```

Comment:
For an abundance of caution, it's better to guard this to onlyOperator and also allow the operator to input an amount. of DOLA to be sent. If the amount is larger than bal, bal is sent instead